assembly/src/tokens/mod.rs

```diff
@@ -255,7 +255,10 @@ impl<'a> Token<'a> {
         assert_eq!(invocation_token, self.parts[0], "not an {invocation_token}");
         match self.num_parts() {
             0 => unreachable!(),
-            1 => Err(ParsingError::missing_param(self)),
+            1 => Err(ParsingError::missing_param(
+                self,
+                "format!(\"{invocation_token}.<procedure_name>\")",
+            )),
```

Comment:
This will just print out a string literal. Instead, the code we want is:
```Rust
 1 => Err(ParsingError::missing_param(
    self,
    format!("{invocation_token}.<procedure_name>"),
)),
```
That is, we want the `format!` macro to format the text correctly.

Reply:
had to do a little more change here, since the result of `format!("malformed instruction '{token}': expected format `{expected_format}`")` would be a String and this was expected to be `&str`.

Let me know if you think this can work.

Reply:
This works, but I wonder if [as_str()](https://doc.rust-lang.org/std/string/struct.String.html#method.as_str) would be a bit cleaner.